lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3443,6 +3451,13 @@ namespace {
         }
       }
 
+      // If this record is declared inside of a nominal type (i.e. a namespace
+      // or another struct), then make sure to add it as a member of that
+      // declaration.
+      if (auto nominalDecl =
+              dyn_cast<NominalTypeDecl>(result->getDeclContext()))
+        nominalDecl->addMember(result);
```

Comment:
Quick comment (not sure I'll have time today to look into it deeper today). Doesn't this collide with this logic for namespaces? https://github.com/apple/swift/blob/5286d9c4ab9c9a1607acee940031d8278fbd465e/lib/ClangImporter/ImportDecl.cpp#L8893

Reply:
Hmm, you might be right. I wonder if this allows us to omit that logic entirely. (That would be nice!) I'll investigate.

Reply:
So, I did some investigation, and you're right. This was the source of the assertion we're hitting. The issue is that this only needs to be here for class template specializations because they're not added in `loadAllMembers`. So, I removed the `addMember` here and moved it to a new `ClassTemplateVisitor` member. Let me know what you think.

Reply:
Also (kind of unrelated) I'm going to put up a patch to refactor the namespace logic in `addAllMembers`.

Reply:
Just FYI @hlopko, `ClassTemplateVisitor` doesn't overlap with #33284. But you might still have merge conflicts.